lib/smart_proxy_ansible/api.rb

```diff
@@ -36,6 +36,36 @@ class Api < Sinatra::Base
         PlaybooksReader.playbooks(params[:playbooks_names]).to_json
       end
 
+      get '/vcs_clone/repo_information' do
+        repo_info = VCSCloner.repo_information(params)
+        status repo_info.status
+        body repo_info.payload.to_json
+      end
+
+      get '/vcs_clone/get_installed' do
+        get_installed = VCSCloner.list_installed_roles
+        status get_installed.status
+        body get_installed.payload.to_json
+      end
+
+      post '/vcs_clone/install' do
+        install = VCSCloner.install(params['repo_info'])
+        status install.status
+        body install.payload.to_json
+      end
+
+      put '/vcs_clone/update' do
+        update = VCSCloner.update(params['repo_info'])
+        status update.status
+        body update.payload.to_json
+      end
+
+      delete '/vcs_clone/delete/:role_name' do
```

Comment:
Similar to my comments on the Foreman PR side (https://github.com/theforeman/foreman_ansible/pull/676), I'd make it a bit more REST. So:
* `GET /vcs_clone/roles` to list
* `POST /vcs_clone/roles` to install
* `PUT /vcs_clone/roles/:name` to update
* `DELETE /vcs_clone/roles/:name` to delete

Reply:
Again, thank you very much for the detailed feedback. Very sorry I forgot to cross-link MRs. Glad you found it anyways!
I added settings for the roles paths am not using plain strings anymore. 
Currently, an exception is thrown if the mutable-roles path does not exist or is not writable (and the feature is enabled), effectively disabling the plugin.
I do however think the verification-API is a good idea. Maybe we could integrate this into the existing `/features` route? Somethink like `/features/active` might be a good idea...